### ISSUE TITLE
Fields are missing after object is created in extended schema models

### ISSUE THREAD

**Issue by wulfricin** [szokodiakos/typegoose#267]
We are using typegoose 5.6.0 and mongoose 5.4.20. After updating from 5.4.1, we lost some functionality where we extend a model with inheritance and the extended models have better defined schemas. 
```
export class BaseModel extends Typegoose {
  @prop({ required: true, _id: false }) tariff: TariffModel;

  public static get Db() {
    return MongooseModelBuilder.build(BaseModel, { collection: 'Bases', versionKey: false });
  }

  public static discriminator(modelName, model) {
    if (BaseModelInstance.discriminators && BaseModelInstance.discriminators[modelName]) {
      return BaseModelInstance.discriminators[modelName];
    }

    return BaseModelInstance.discriminator(modelName, model.schema);
  }
}

const schemaOptions = { collection: 'Bases', timestamps: true };
// tslint:disable-next-line:variable-name
const BaseModelInstance = new BaseModel().getModelForClass(BaseModel, { schemaOptions });

export class ExtendedBaseModel extends Typegoose {
  @prop({ required: true, _id: false }) tariff: ExtendedTariffModel;

  public static get Db() {
    const model = MongooseModelBuilder.build(ExtendedBaseModel, { _id: false });

    return BaseModel.discriminator(`Extended-Base`, model);
  }
}


export class TariffModel extends Typegoose {
  @prop({ required: true }) name: string;
  @prop({ required: true }) logo: string;
  @prop({ required: false, _id: false }) productSpecificData: any;
}

export class ExtendedTariffDataModel {
  @prop({ required: false }) externalLink?: string;
}

export class ExtendedTariffModel extends TariffModel {
  @prop({ required: true, _id: false }) productSpecificData: ExtendedTariffDataModel;
}
```

When we save this model using the following
```
 const objectToCreate = {
    tariff: {
    name: 'tariff',
    logo: 'logo.png',
    productSpecificData: {
      externalLink: 'some link'
    }
  }
};
const extendedBaseInstance= new ExtendedBaseModel.Db(objectToCreate);
const savedObject= await extendedBaseInstance.save() as ExtendedBase;
```

the savedObject is missing the fields tariff.name and tariff.logo eventhough the objectToCreate has the fields in it. The productSpecificData field is saved when using this code. Is this due to some changes in mongoose or typegoose?

**Comment by hasezoey** [comment]
this can be closed in favor of #332 and marked as "stale" / ("outdated"?)
@Ben305 

if someone still has this issue, please open a new issue with "minimal" code

**Comment by hasezoey** [comment]
@Ben305 Bump to close
